client/src/pages/private/participant-details/constants.js

```diff
@@ -0,0 +1,105 @@
+import pick from 'lodash/pick';
+
+import { fetchParticipant, psi, getAllSites } from '../../../services';
+import { addYearToDate, dayUtils } from '../../../utils';
+
+// Key Map
+export const keyLabelMap = {
+  fullName: 'Full Name',
+  phoneNumber: 'Phone Number',
+  emailAddress: 'Email Address',
+  interested: 'Program Interest',
+  preferredLocation: 'Preferred Location',
+  postalCodeFsa: 'Postal Code FSA',
+  cohortName: 'Cohort / PSI',
+  postHireStatusLabel: 'Graduation Status',
+};
+
+// ===   Functions  ===
+// Map Ros Data
+const mapRosData = ({ data = {}, rosSite = {} }) => {
+  const { date, startDate } = data;
+  const { siteName, healthAuthority } = rosSite;
+  return {
+    date: dayUtils(date).format('MMM DD, YYYY'),
+    endDate: addYearToDate(date).format('MMM DD, YYYY'),
+    startDate: startDate ? dayUtils(startDate).format('MMM DD, YYYY') : undefined,
+    siteName,
+    healthAuthority,
+  };
+};
+
+const getInterestLabel = (isInterested) => {
+  switch (isInterested) {
+    case 'yes':
+      return 'Interested';
+    case 'no':
+      return 'Withdrawn';
+    default:
+      return isInterested;
+  }
+};
+
+export const getLinkLabel = (pageName) => {
+  return pageName === 'participant' ? 'Participant' : 'Site View';
+};
+
+// Display Data
+export const displayData = (inputData) => ({
+  ...pick(inputData, Object.keys(keyLabelMap)),
+  fullName: `${inputData.firstName} ${inputData.lastName}`,
+  interested: getInterestLabel(inputData.interested),
+  ros:
+    inputData.rosStatus && Object.keys(inputData.rosStatus).length
+      ? mapRosData(inputData.rosStatus)
+      : null,
+});
+
+// Helper
+export const fetchData = ({
+  setParticipant,
+  setActualParticipant,
+  setPSIList,
+  id,
+  setError,
+  setDisableAssign,
+  setAllSites,
+}) => {
+  fetchParticipant({ id })
```

Comment:
This might be more readable with async/await, was there a reason to use promise chaining?

Reply:
I'm not certain about the reasoning behind using promises in this case. It could be because it's called in `useEffect` which doesn't always tolerate async/await